### PR TITLE
🌱 Add indexers.ByPathAndName helper

### PR DESCRIPTION
**Pull request by ncdc**
## Summary
Add indexers.ByPathAndName helper to reduce lots of copying 😄 

## Related issue(s)

Fixes #

### CODE REVIEW

File: pkg/indexers/indexers.go [hunk 1]
```diff
@@ -124,3 +125,18 @@ func ByIndex[T runtime.Object](indexer cache.Indexer, indexName, indexValue stri
 
 	return ret, nil
 }
+
+// ByPathAndName returns the instance of T from the indexer with the matching path and name.
```
Comment (sttts) [inline pkg/indexers/indexers.go:129]: mention that this is partial and depends on the path annotation.

Also: it matches both a canonical path **and** the cluster name.

Comment (stevekuznetsov) [reply pkg/indexers/indexers.go:129]: Full path or any permutation?

**Comment by ncdc** [comment]
Flake #2501

File: pkg/indexers/indexers.go (hunk 1 above)
Comment (ncdc) [reply pkg/indexers/indexers.go]: Updated the comment; PTAL

**Review by stevekuznetsov — APPROVED** [review]
Seems fine - how do I know which types will have a working annotation?

**Pull request merged**